packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/pages/edit/page-list.astro

```diff
@@ -0,0 +1,113 @@
+import Config from 'virtual:studiocms/config';
+import { authHelper, type Locals } from 'studiocms:helpers';
+import { FormattedDate } from 'studiocms:components';
+import { StudioCMSRoutes, getEditRoute, getDeleteRoute } from 'studiocms-dashboard:routeMap';
+import type { SiteConfigResponse, pageDataReponse } from 'studiocms:components';
+
+if (Config.includedIntegrations.useInoxSitemap) {
+    import('sitemap-ext:config').then((sitemap) => {
+        sitemap.default(false)
+    })
+}
+
+const { dashboardConfig: { developerConfig: { testingAndDemoMode } } } = Config;
+
+const { 
+    authLinks: { 
+        loginURL 
+    }, 
+    mainLinks: { 
+        userProfile,
+        dashboardIndex,
+        pageEdit,
+    } 
+} = StudioCMSRoutes;
+
+const contextConfig: SiteConfigResponse = await db.select().from(SiteConfig).where(eq(SiteConfig.id, 1)).get();
+const pages: pageDataReponse[] = await db.select().from(PageData).orderBy(asc(PageData.id));
+const { permissionLevel } = await authHelper(Astro.locals as Locals);
+
+// If the user is not logged in, redirect to the login page
+if ( !testingAndDemoMode ) {
+    if ( permissionLevel === "unknown" ) {
+        console.log('User is not logged in. Redirecting to login page.');
+        return Astro.redirect(loginURL);
+    }
+    if ( permissionLevel !== 'admin' && permissionLevel !== 'editor' ) {
+        console.log('User is not an admin or editor. Redirecting to profile page.');
+        return Astro.redirect(userProfile);
+    }
+}
+
+const pageTitle = `Existing Pages | ${contextConfig.title}`;
+---
+
+<Layout 
+    sideBarActiveItemID="edit-pages" 
+    title={pageTitle}
+    description={contextConfig.description} >
+
+    <div class="flex grow w-full pb-4">
+        <sl-breadcrumb>
+            <span slot="separator">/</span>
+            <sl-breadcrumb-item href={dashboardIndex}>Dashboard
+                <sl-icon slot="prefix" name="columns-gap"></sl-icon>
+            </sl-breadcrumb-item>
+            <sl-breadcrumb-item href={pageEdit}>Existing Pages</sl-breadcrumb-item>
+        </sl-breadcrumb>
+    </div>
+
+    <div class="container max-w-[100vw] lg:max-w-[1800px] min-w-[75vw]">
+
+        <div class="text-5xl font-bold py-8">Existing Pages</div>
+
+        <div class="py-2">
+            <div class="relative">
+                <table class="w-full text-left rtl:text-right">
+                    <thead class="uppercase bg-base-300 text-inherit">
+                        <tr>
+                            <th scope="col" class="px-6 py-3">Page Title</th>
+                            <th scope="col" class="px-6 py-3">Publish Date</th>
+                            <th scope="col" class="px-6 py-3">Edit</th>
+                            <th scope="col" class="px-6 py-3">Delete</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        { 
+                            pages.map(async (page) => {
```

Comment:
It only works because SSR... lol in SSG this would fight me all day long... but because im awaiting using a async function in the thing i have to make the pages.map async